tests/test_implicit_networks.cpp

```diff
@@ -808,3 +810,405 @@ TEST_CASE("material interface on a failed example", "[MI][examples][!shouldfail]
+        REQUIRE(success);
+        csg_file = std::string(TEST_FILE) + "/csg_tree_1.json";
+        csg(iso_pts,
+                   iso_faces,
+                   patches,
+                   patch_function_label,
+                   iso_edges,
+                   chains,
+                   non_manifold_edges_of_vert,
+                   shells,
+                   arrangement_cells,
+                   cell_function_label,
+                   lambda);
+        size_t corners_count = 0;
+        for (size_t i = 0; i < non_manifold_edges_of_vert.size(); i++) {
+            if (non_manifold_edges_of_vert[i].size() > 2) {
+                corners_count++;
+            }
+        }
+        std::cout << "corners: " << corners_count << std::endl;
+        std::cout << "cell labels: " << std::endl;
+        // check
+        REQUIRE(patches.size() == 0);
+        REQUIRE(chains.size() == 0);
+        REQUIRE(corners_count == 0);
+        REQUIRE(shells.size() == 0);
+    }
+    
+    SECTION("three spheres config 2") {
+        // compute function values on tet grid vertices
+        size_t n_pts = pts.size();
+        size_t n_func = 3;
+        funcVals.resize(n_pts, n_func);
+        size_t func_id;
+        if (!load_functions(std::string(TEST_FILE) + "/3-sphere-2.json", pts, funcVals)) {
+            throw std::runtime_error("ERROR: Failed to load functions.");
+        }
+        // compute material interface
+        bool success = implicit_arrangement(
+                robust_test,
+                use_lookup,
+                use_secondary_lookup,
+                use_topo_ray_shooting,
+                //
+                pts, tets, funcVals,
+                //
+                iso_pts,iso_faces,patches,
+                patch_function_label,
+                iso_edges,chains,
+                non_manifold_edges_of_vert,
+                shells,arrangement_cells,cell_function_label,
+                timing_labels,timings,
+                stats_labels,stats);
+        REQUIRE(success);
+        csg_file = std::string(TEST_FILE) + "/csg_tree_2.json";
+        csg(iso_pts,
+                   iso_faces,
+                   patches,
+                   patch_function_label,
+                   iso_edges,
+                   chains,
+                   non_manifold_edges_of_vert,
+                   shells,
+                   arrangement_cells,
+                   cell_function_label,
+                   lambda);
+        size_t corners_count = 0;
+        for (size_t i = 0; i < non_manifold_edges_of_vert.size(); i++) {
+            if (non_manifold_edges_of_vert[i].size() > 2) {
+                corners_count++;
+            }
+        }
+        std::cout << "corners: " << corners_count << std::endl;
+        std::cout << "cell labels: " << std::endl;
+        // check
+        REQUIRE(patches.size() == 4);
+        REQUIRE(chains.size() == 2);
+        REQUIRE(corners_count == 0);
+        REQUIRE(shells.size() == 2);
```

Comment:
Is this correct? Is the following visualization corresponding to this test?
<img width="791" alt="image" src="https://github.com/duxingyi-charles/Robust-Implicit-Surface-Networks/assets/3606672/f95350ed-32cf-4815-b555-1c44eec72fdc">